src/h5web/providers/models.ts

```diff
@@ -57,32 +57,32 @@ export enum MyHDF5EntityKind {
 
 export interface MyHDF5Entity {
   uid: string;
-  id?: HDF5Id;
   name: string;
   kind: MyHDF5EntityKind;
   parents: MyHDF5Group[];
   attributes: HDF5Attribute[];
   rawLink?: HDF5Link;
 }
 
-export interface MyHDF5Group extends MyHDF5Entity {
+export interface MyHDF5ResolvedEntity extends MyHDF5Entity {
   id: HDF5Id;
+}
```

Comment:
Introducing the concept of a "resolved" entity (as opposed to unresolved links). This allows me to add a guard called `isResolved`, useful when I need to access an entity's `id`.

Reply:
Should I keep the changes as is for now or would you prefer I revert?

Reply:
You can merge as it is. I will make a thorough review of all of this when I have time.